Realm/RLMSyncSubscription.h

```diff
@@ -0,0 +1,336 @@
+    /// find the error calling error in the subscription set to get a description of the error. You can
+    /// still use the current subscription set to write a subscription.
+    RLMSyncSubscriptionStateError,
+    /// The subscription is persisted locally but not yet processed by the server, which means
+    /// the server hasn't yet returned all the data that matched the updated subscription queries.
+    RLMSyncSubscriptionStatePending,
+    /// The subscription set has been super-ceded by an updated one, this typically means that
+    /// someone is trying to write a subscription on a different instance of the subscription set.
+    /// You should not use a super-ceded subscription set and instead obtain a new instance of
+    /// the subscription set to write a subscription.
+    RLMSyncSubscriptionStateSuperceded
+};
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface RLMSyncSubscription : NSObject
+
+/// Identifier of the subscription.
+@property (nonatomic, readonly) RLMObjectId *identifier;
+
+/// Name of the subscription. If not specified it will return nil.
+@property (nonatomic, readonly, nullable) NSString *name;
+
+/// When the subscription was created. Recorded automatically.
+@property (nonatomic, readonly) NSDate *createdAt;
+
+/// When the subscription was last updated. Recorded automatically.
+@property (nonatomic, readonly) NSDate *updatedAt;
+
+/**
+ Updates a Flexible Sync's subscription query with an allowed query which will be used to bootstrap data
+ from the server when committed.
+
+ @warning This method may only be called during a write subscription block.
+
+ @param objectClassName The class name for the model class to be queried.
+ @param predicateFormat A predicate format string, optionally followed by a variable number of arguments.
+ */
+- (void)updateSubscriptionWithClassName:(NSString *)objectClassName
+                                  where:(NSString *)predicateFormat, ...;
+
+/// :nodoc:
+- (void)updateSubscriptionWithClassName:(NSString *)objectClassName
+                                  where:(NSString *)predicateFormat
+                                   args:(va_list)args;
+
+/**
+ Updates a Flexible Sync's subscription query with an allowed query which will be used to bootstrap data
+ from the server when committed.
+
+ @warning This method may only be called during a write subscription block.
+
+ @param objectClassName The class name for the model class to be queried.
+ @param predicate The predicate with which to filter the objects on the server.
+ */
+- (void)updateSubscriptionWithClassName:(NSString *)objectClassName
+                              predicate:(NSPredicate *)predicate;
+
+@end
+
+@interface RLMSyncSubscriptionSet : NSObject <NSFastEnumeration>
+
+/// The number of subscriptions in the subscription set.
+@property (readonly) NSUInteger count;
+
+/// Gets the error associated to the subscription set, this will return a no nil in case the current
```

Comment:
This comma should be a period. Same thing in the next few docs as well.

"This will be non-nil in case"